packages/flutter/lib/src/rendering/block.dart

```diff
@@ -187,6 +189,8 @@ class RenderBlock extends RenderBox
       case Axis.vertical:
         return _getIntrinsicCrossAxis((RenderBox child) => child.getMinIntrinsicWidth(height));
     }
+    assert(mainAxis != null);
```

Comment:
The way you have it is fine IMHO. I think I would have put the conditional assert above the switch or the false assert below the switch but it really is neither here nor there and, again, what you have is totally fine IMHO.